File: api/ocr.py
import os
import requests

def handler(request):
    if request.method != "POST":
        return {
            "statusCode": 405,
            "body": "Method Not Allowed"
        }

    OCR_API_KEY = os.environ.get("OCR_API_KEY")

    file = request.files.get("file")

    if not file:
        return {
            "statusCode": 400,
            "body": "No file uploaded"
        }

    payload = {
        "apikey": OCR_API_KEY,
        "language": "eng",
        "ocrengine": 3,
    }

    response = requests.post(
        "https://api.ocr.space/parse/image",
        files={"file": (file.filename, file.stream, file.mimetype)},
        data=payload
    )

    result = response.json()

    try:
        text = result["ParsedResults"][0]["ParsedText"]
    except:
        text = "Error extracting text"

    return {
        "statusCode": 200,
        "headers": {"Content-Type": "application/json"},
        "body": text
    }

File: api/ocr.js
export const config = {
  api: {
    bodyParser: false,
  },
};

import formidable from "formidable";
import fs from "fs";
import fetch from "node-fetch";
import FormData from "form-data";

export default async function handler(req, res) {
  if (req.method !== "POST") {
    return res.status(405).send("Method Not Allowed");
  }

  const form = new formidable.IncomingForm();

  form.parse(req, async (err, fields, files) => {
    if (err) {
      return res.status(500).send("Error parsing file");
    }

    const file = files.file;

    if (!file) {
      return res.status(400).send("No file uploaded");
    }

    const formData = new FormData();
    formData.append("file", fs.createReadStream(file.filepath));
    formData.append("apikey", process.env.OCR_API_KEY);
    formData.append("language", "eng");
    formData.append("ocrengine", "3");

    try {
      const response = await fetch("https://api.ocr.space/parse/image", {
        method: "POST",
        body: formData,
      });

      const data = await response.json();

      const text =
        data?.ParsedResults?.[0]?.ParsedText || "No text found";

      res.status(200).send(text);
    } catch (error) {
      res.status(500).send("OCR API Error");
    }
  });
}
